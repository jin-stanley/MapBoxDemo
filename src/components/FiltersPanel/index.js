/***
 * Using for display filter form
 */
import React, { useState, useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";

import SettingWrapper from "../SettingWrapper";
import {
  TextField,
  MenuItem,
  Select,
  Button,
  Divider,
  Slider,
} from "@material-ui/core";

const FiltersPanel = (props) => {
  const classes = useStyles();

  // Here set form init value
  useEffect(() => {
    setFormValue(props.filters);
  }, [props.filters]);

  const [formValue, setFormValue] = useState({
    title_contains: "",
    stage: "",
    dev_type: "",
    ownership: "",
    value: [0, 100],
    floor_area: [0, 100],
    site_area: [0, 100],
  });

  const handleChangeValue = (e, field) => {
    const newFormValue = formValue;
    newFormValue[field] = e.target.value;
    setFormValue({ ...newFormValue });
  };

  const handleSubmite = () => {
    props.setFilters(formValue);
    props.handleCloseSetting();
  };

  const handleCancle = () => {
    props.handleCloseSetting();
  };

  return (
    <SettingWrapper
      name="Filters"
      handleCloseSetting={props.handleCloseSetting}
    >
      <div className={classes.container}>
        <div className={classes.controllerWrapper}>
          <div className={classes.controllerName}>Title Contains: </div>
          <TextField
            value={formValue.title_contains}
            className={classes.controller}
            margin="dense"
            variant="outlined"
            onChange={(e) => handleChangeValue(e, "title_contains")}
          />
        </div>

        <div className={classes.controllerWrapper}>
          <div className={classes.controllerName}>Stage: </div>
          <Select
            className={classes.controller}
            margin="dense"
            variant="outlined"
            value={formValue.stage}
            onChange={(e) => handleChangeValue(e, "stage")}
          >
            <MenuItem value="">
              <em>None</em>
            </MenuItem>
            {!!props.formSelects["Stages"] &&
              props.formSelects.Stages.map((stage, key) => {
                return (
                  <MenuItem key={key} value={stage}>
                    {stage}
                  </MenuItem>
                );
              })}
          </Select>
        </div>

        <div className={classes.controllerWrapper}>
          <div className={classes.controllerName}>Ownership: </div>
          <Select
            className={classes.controller}
            margin="dense"
            variant="outlined"
            value={formValue.ownership}
            onChange={(e) => handleChangeValue(e, "ownership")}
          >
            <MenuItem value="">
              <em>None</em>
            </MenuItem>
            {!!props.formSelects["Ownerships"] &&
              props.formSelects.Ownerships.map((ownership, key) => {
                return (
                  <MenuItem key={key} value={ownership}>
                    {ownership}
                  </MenuItem>
                );
              })}
          </Select>
        </div>

        <div className={classes.controllerWrapper}>
          <div className={classes.controllerName}>Dev. Type: </div>
          <Select
            className={classes.controller}
            margin="dense"
            variant="outlined"
            value={formValue.dev_type}
            onChange={(e) => handleChangeValue(e, "dev_type")}
          >
            <MenuItem value="">
              <em>None</em>
            </MenuItem>
            {!!props.formSelects["DevTypes"] &&
              props.formSelects.DevTypes.map((devType, key) => {
                return (
                  <MenuItem key={key} value={devType}>
                    {devType}
                  </MenuItem>
                );
              })}
          </Select>
        </div>

        <div className={classes.controllerWrapper}>
          <div className={classes.controllerName}>
            Value:
            <br />
            <span>({`${formValue.value[0]} - ${formValue.value[1]}`})</span>
          </div>
          <div className={classes.controller}>
            <Slider
              value={formValue.value}
              onChange={(e, newValue) =>
                setFormValue({ ...formValue, value: newValue })
              }
              valueLabelDisplay="auto"
              aria-labelledby="range-slider"
              min={
                !!props.formSelects["ValueRange"]
                  ? props.formSelects["ValueRange"][0]
                  : 0
              }
              step={500}
              max={
                !!props.formSelects["ValueRange"]
                  ? props.formSelects["ValueRange"][1]
                  : 1000
              }
            />
          </div>
        </div>

        <div className={classes.controllerWrapper}>
          <div className={classes.controllerName}>
            Floor Area:
            <br />
            <span>
              ({`${formValue.floor_area[0]} - ${formValue.floor_area[1]}`})
            </span>
          </div>
          <div className={classes.controller}>
            <Slider
              value={formValue.floor_area}
              onChange={(e, newValue) =>
                setFormValue({ ...formValue, floor_area: newValue })
              }
              valueLabelDisplay="auto"
              aria-labelledby="range-slider"
              min={
                !!props.formSelects["FloorAreaRange"]
                  ? props.formSelects["FloorAreaRange"][0]
                  : 1000
              }
              step={500}
              max={
                !!props.formSelects["FloorAreaRange"]
                  ? props.formSelects["FloorAreaRange"][1]
                  : 1000
              }
            />
          </div>
        </div>

        <div className={classes.controllerWrapper}>
          <div className={classes.controllerName}>
            Site Area:
            <br />
            <span>
              ({`${formValue.site_area[0]} - ${formValue.site_area[1]}`})
            </span>
          </div>
          <div className={classes.controller}>
            <Slider
              value={formValue.site_area}
              onChange={(e, newValue) =>
                setFormValue({ ...formValue, site_area: newValue })
              }
              valueLabelDisplay="auto"
              aria-labelledby="range-slider"
              min={
                !!props.formSelects["SiteAreaRange"]
                  ? props.formSelects["SiteAreaRange"][0]
                  : 1000
              }
              step={500}
              max={
                !!props.formSelects["SiteAreaRange"]
                  ? props.formSelects["SiteAreaRange"][1]
                  : 1000
              }
            />
          </div>
        </div>
      </div>

      <Divider />

      <div className={classes.ButtonWrapper}>
        <Button variant="contained" color="secondary" onClick={handleCancle}>
          Cancel
        </Button>
        <Button variant="contained" color="primary" onClick={handleSubmite}>
          Confirm
        </Button>
      </div>
    </SettingWrapper>
  );
};

const useStyles = makeStyles({
  container: {
    height: "320px",
    minWidth: "240px",
    position: "relative",
    flex: "1 1 0",
    overflow: "auto",
  },
  controllerWrapper: {
    display: "flex",
    alignItems: "center",
    width: "100%",
    margin: "12px 0",
  },
  controllerName: {
    width: "30%",
    fontSize: "12px",
    fontWeight: 800,
    "& span": {
      fontSize: "10px",
    },
  },
  controller: {
    width: "70%",
    margin: "8px",
  },
  ButtonWrapper: {
    display: "flex",
    alignItems: "center",
    margin: "8px 0",
    justifyContent: "space-between",
  },
});

export default FiltersPanel;
