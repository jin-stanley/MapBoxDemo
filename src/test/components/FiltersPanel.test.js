import React from "react";
import renderer from "react-test-renderer";

import FiltersPanel from "../../components/FiltersPanel";

it("FiltersPanel can render success", () => {
  const tree = renderer
    .create(
      <FiltersPanel
        formSelects={{
          DevTypes: ["a", "b"],
          FloorAreaRange: [1, 2],
          Ownerships: ["a", "b"],
          Stages: ["a", "b"],
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
