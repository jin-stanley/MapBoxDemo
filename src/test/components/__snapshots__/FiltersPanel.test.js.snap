// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersPanel can render success 1`] = `
<div
  className="makeStyles-container-6"
>
  <div
    className="makeStyles-header-7"
  >
    <p>
      Filters
    </p>
    <svg
      className="makeStyles-closeIcon-9"
      onClick={[Function]}
    >
      x.svg
    </svg>
  </div>
  <div
    className="makeStyles-content-8"
  >
    <div
      className="makeStyles-container-1"
    >
      <div
        className="makeStyles-controllerWrapper-2"
      >
        <div
          className="makeStyles-controllerName-3"
        >
          Title Contains: 
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-controller-4 MuiFormControl-marginDense"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-11"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="makeStyles-controllerWrapper-2"
      >
        <div
          className="makeStyles-controllerName-3"
        >
          Stage: 
        </div>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root makeStyles-controller-4 MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value=""
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-11"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="makeStyles-controllerWrapper-2"
      >
        <div
          className="makeStyles-controllerName-3"
        >
          Ownership: 
        </div>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root makeStyles-controller-4 MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value=""
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-11"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="makeStyles-controllerWrapper-2"
      >
        <div
          className="makeStyles-controllerName-3"
        >
          Dev. Type: 
        </div>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root makeStyles-controller-4 MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value=""
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-11"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="makeStyles-controllerWrapper-2"
      >
        <div
          className="makeStyles-controllerName-3"
        >
          Value:
          <br />
          <span>
            (
            0 - 100
            )
          </span>
        </div>
        <div
          className="makeStyles-controller-4"
        >
          <span
            className="MuiSlider-root MuiSlider-colorPrimary"
            onMouseDown={[Function]}
          >
            <span
              className="MuiSlider-rail"
            />
            <span
              className="MuiSlider-track"
              style={
                Object {
                  "left": "0%",
                  "width": "10%",
                }
              }
            />
            <input
              type="hidden"
              value="0,100"
            />
            <span
              aria-labelledby="range-slider"
              aria-orientation="horizontal"
              aria-valuemax={1000}
              aria-valuemin={0}
              aria-valuenow={0}
              className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-14"
              data-index={0}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              role="slider"
              style={
                Object {
                  "left": "0%",
                }
              }
              tabIndex={0}
            >
              <span
                className="PrivateValueLabel-offset-16 MuiSlider-valueLabel"
              >
                <span
                  className="PrivateValueLabel-circle-17"
                >
                  <span
                    className="PrivateValueLabel-label-18"
                  >
                    0
                  </span>
                </span>
              </span>
            </span>
            <span
              aria-labelledby="range-slider"
              aria-orientation="horizontal"
              aria-valuemax={1000}
              aria-valuemin={0}
              aria-valuenow={100}
              className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-14"
              data-index={1}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              role="slider"
              style={
                Object {
                  "left": "10%",
                }
              }
              tabIndex={0}
            >
              <span
                className="PrivateValueLabel-offset-16 MuiSlider-valueLabel"
              >
                <span
                  className="PrivateValueLabel-circle-17"
                >
                  <span
                    className="PrivateValueLabel-label-18"
                  >
                    100
                  </span>
                </span>
              </span>
            </span>
          </span>
        </div>
      </div>
      <div
        className="makeStyles-controllerWrapper-2"
      >
        <div
          className="makeStyles-controllerName-3"
        >
          Floor Area:
          <br />
          <span>
            (
            0 - 100
            )
          </span>
        </div>
        <div
          className="makeStyles-controller-4"
        >
          <span
            className="MuiSlider-root MuiSlider-colorPrimary"
            onMouseDown={[Function]}
          >
            <span
              className="MuiSlider-rail"
            />
            <span
              className="MuiSlider-track"
              style={
                Object {
                  "left": "0%",
                  "width": "100%",
                }
              }
            />
            <input
              type="hidden"
              value="1,2"
            />
            <span
              aria-labelledby="range-slider"
              aria-orientation="horizontal"
              aria-valuemax={2}
              aria-valuemin={1}
              aria-valuenow={1}
              className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-14"
              data-index={0}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              role="slider"
              style={
                Object {
                  "left": "0%",
                }
              }
              tabIndex={0}
            >
              <span
                className="PrivateValueLabel-offset-16 MuiSlider-valueLabel"
              >
                <span
                  className="PrivateValueLabel-circle-17"
                >
                  <span
                    className="PrivateValueLabel-label-18"
                  >
                    1
                  </span>
                </span>
              </span>
            </span>
            <span
              aria-labelledby="range-slider"
              aria-orientation="horizontal"
              aria-valuemax={2}
              aria-valuemin={1}
              aria-valuenow={2}
              className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-14"
              data-index={1}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              role="slider"
              style={
                Object {
                  "left": "100%",
                }
              }
              tabIndex={0}
            >
              <span
                className="PrivateValueLabel-offset-16 MuiSlider-valueLabel"
              >
                <span
                  className="PrivateValueLabel-circle-17"
                >
                  <span
                    className="PrivateValueLabel-label-18"
                  >
                    2
                  </span>
                </span>
              </span>
            </span>
          </span>
        </div>
      </div>
      <div
        className="makeStyles-controllerWrapper-2"
      >
        <div
          className="makeStyles-controllerName-3"
        >
          Site Area:
          <br />
          <span>
            (
            0 - 100
            )
          </span>
        </div>
        <div
          className="makeStyles-controller-4"
        >
          <span
            className="MuiSlider-root MuiSlider-colorPrimary"
            onMouseDown={[Function]}
          >
            <span
              className="MuiSlider-rail"
            />
            <span
              className="MuiSlider-track"
              style={
                Object {
                  "left": "NaN%",
                  "width": "NaN%",
                }
              }
            />
            <input
              type="hidden"
              value="1000,1000"
            />
            <span
              aria-labelledby="range-slider"
              aria-orientation="horizontal"
              aria-valuemax={1000}
              aria-valuemin={1000}
              aria-valuenow={1000}
              className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-14"
              data-index={0}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              role="slider"
              style={
                Object {
                  "left": "NaN%",
                }
              }
              tabIndex={0}
            >
              <span
                className="PrivateValueLabel-offset-16 MuiSlider-valueLabel"
              >
                <span
                  className="PrivateValueLabel-circle-17"
                >
                  <span
                    className="PrivateValueLabel-label-18"
                  >
                    1000
                  </span>
                </span>
              </span>
            </span>
            <span
              aria-labelledby="range-slider"
              aria-orientation="horizontal"
              aria-valuemax={1000}
              aria-valuemin={1000}
              aria-valuenow={1000}
              className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-14"
              data-index={1}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              role="slider"
              style={
                Object {
                  "left": "NaN%",
                }
              }
              tabIndex={0}
            >
              <span
                className="PrivateValueLabel-offset-16 MuiSlider-valueLabel"
              >
                <span
                  className="PrivateValueLabel-circle-17"
                >
                  <span
                    className="PrivateValueLabel-label-18"
                  >
                    1000
                  </span>
                </span>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      className="makeStyles-ButtonWrapper-5"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Cancel
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Confirm
        </span>
      </button>
    </div>
  </div>
</div>
`;
